### PR TITLE
Improve messages

### PR DESCRIPTION
**Pull request by acostami**
### Proposal of a new message format - #11

`assert.ok(true) file_path.js:[line]:[column]`

#### Example

`assert.ok(true) shell/tests/acceptance/some-test.js:23:2`

* Add file path to message
* Add line and column to message
* Make messages using the original line

### CODE REVIEW

**Comment by fedekau** [comment]
@acostami I have configured the repository so that `master` is a protected branch, and anything going into it must pass all the checks, take a look at codeclimate and try to DRY out the code accordingly.

File: node-tests/tests-transform-filter-test.js [hunk 1]
```diff
@@ -30,6 +30,7 @@ describe('transform test files', function() {
 
     return build('fixtures/dummy-test-folder').then(function(results) {
       files(results).forEach(function(file) {
+        console.log(file);
```
Comment (fedekau) [inline node-tests/tests-transform-filter-test.js]: Please remove this :)

**Comment by fedekau** [comment]
I will close this, so we continue the conversation in only one place #17 :)

File: README.md [hunk 2]
```diff
@@ -30,6 +30,29 @@ test('it works', function(assert) {
 But with **ember-qunit-nice-errors** the message is way nicer!
 ![Test failed output with addon](https://github.com/wyeworks/ember-qunit-nice-errors/raw/gh-pages/images/after.png)
 
+## Configuration
+
+### showFileInfo
+
+If you want your error messages to include the original test file, line and column where the fail assertion is, just add the following configuration on your `config/environment.js` file:
```
Comment (fedekau) [inline README.md]: `where the fail assertion is` => `where the failed assertion is`

File: node-tests/fixtures/original/unit/with-file-line.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+module('Unit | Helper | test helper');
+
+//Replace this with your real tests.
+test('it works', function(assert) {
+  var result = false;
+
+  assert.ok(result);
+  assert.equal(1, 3);
+
+     assert.notOk(1===1);
```
Comment (fedekau) [inline node-tests/fixtures/original/unit/with-file-line.js]: Is this on purpose (the extra spaces)?

File: node-tests/fixtures/transformed/integration/with-file/example-test.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+import { testHelper } from 'shell/helpers/test-helper';
+import { module, test } from 'qunit';
+
+module('Unit | Helper | test helper');
+
+test('it works', function(assert) {
+  assert.ok(result, 'assert.ok(result) at example-test.js:7:2');
+  assert.notOk(1===1, 'assert.notOk(1===1) at example-test.js:8:2');
+     assert.equal(1, 2, 'assert.equal(1, 2) at example-test.js:9:5');
```
Comment (fedekau) [inline node-tests/fixtures/transformed/integration/with-file/example-test.js]: Is this on purpose (the extra spaces)?

Comment (san650) [inline node-tests/fixtures/transformed/integration/with-file/example-test.js]: I guess we're testing the column position `:5` here.

Comment (sdebarros) [inline node-tests/fixtures/transformed/integration/with-file/example-test.js]: @san650 is correct

File: index.js [hunk 5]
```diff
@@ -6,9 +6,16 @@ var TestTransformFilter = require('./lib/tests-transform-filter');
 module.exports = {
   name: 'ember-qunit-nice-errors',
 
+  included: function(app) {
+    this._super.included.apply(this, arguments);
+
+    this.addonConfig =
+      app.project.config(app.env)['ember-qunit-nice-errors'] || {};
```
Comment (san650) [inline index.js]: There's a known issue we need cover from

See https://github.com/ember-cli/ember-cli/issues/3718
and a workaround https://github.com/samselikoff/ember-cli-mirage/blob/master/index.js#L24-L27

File: lib/tests-transform-filter.js [hunk 6]
```diff
@@ -5,17 +5,19 @@ var Filter = require('broccoli-filter');
 var transform = require('./transform-qunit-assertions');
 
 function TestTransformFilter(inputNode, options) {
-  options = options || {};
-  Filter.call(this, inputNode, options);
+  this.options = options || {};
```
Comment (san650) [inline lib/tests-transform-filter.js]: Looking at `index.js` we can assume that options is an object.

File: lib/transform-qunit-assertions.js [hunk 7]
```diff
@@ -6,27 +6,43 @@ var recast = require('recast');
...
-    node.arguments.push(builders.literal(prefix + ' ' + str));
+function addAssertionMessage(node, options) {
+  var str = recast.print(node).code;
+  var extra = '';
+
+  if (options && options.file) {
+    extra = ' at ' + location(node, options.file);
   }
+
+  node.arguments.push(builders.literal(str + extra));
```
Comment (san650) [inline lib/transform-qunit-assertions.js]: Just append to `str`

```js
var str = ...

if (options && options.file) {
  str += ' at ' + ....;
}

node.arguments.push(builders.literal(str));
```

Comment (sdebarros) [inline lib/transform-qunit-assertions.js]: 👍

File: lib/tests-transform-filter.js (hunk 6 above)
Comment (sdebarros) [inline lib/tests-transform-filter.js]: 👍

**Pull request merged**